src/node/stage/approved.rs

```diff
@@ -1381,113 +1519,58 @@ impl Approved {
-                    AccumulatingEvent::Relocate(ref details)
-                    | AccumulatingEvent::RelocatePrepare(ref details, _) => {
-                        our_prefix.matches(details.pub_id.name())
-                    }
-                    // Drop: no longer relevant after prefix change.
-                    AccumulatingEvent::Genesis { .. }
-                    | AccumulatingEvent::StartDkg(_)
-                    | AccumulatingEvent::DkgResult { .. }
-                    | AccumulatingEvent::ParsecPrune => false,
-
-                    // Keep: Additional signatures for neighbours for sec-msg-relay.
-                    AccumulatingEvent::SectionInfo(ref elders_info, _)
-                    | AccumulatingEvent::NeighbourInfo(ref elders_info, _) => {
-                        our_prefix.is_neighbour(&elders_info.prefix)
-                    }
-
-                    // Only revote if the recipient is still our neighbour
-                    AccumulatingEvent::SendNeighbourInfo { ref dst, .. } => {
-                        self.shared_state.sections.is_in_neighbour(dst)
-                    }
-
-                    // Keep: Still relevant after prefix change.
-                    AccumulatingEvent::TheirKeyInfo { .. }
-                    | AccumulatingEvent::TheirKnowledge { .. }
-                    | AccumulatingEvent::User(_) => true,
-                }
-            })
-            .collect();
-
         Ok(ParsecResetData {
             genesis_prefix_info,
             to_vote_again,
-            to_process,
         })
     }
 
-    fn process_post_reset_events(&mut self, core: &mut Core, to_process: BTreeSet<NetworkEvent>) {
-        to_process.iter().for_each(|event| match &event.payload {
-            AccumulatingEvent::Genesis { .. }
-            | AccumulatingEvent::Offline(_)
-            | AccumulatingEvent::StartDkg(_)
-            | AccumulatingEvent::DkgResult { .. }
-            | AccumulatingEvent::ParsecPrune
-            | AccumulatingEvent::Relocate(_)
-            | AccumulatingEvent::RelocatePrepare(_, _)
-            | AccumulatingEvent::SectionInfo(_, _)
-            | AccumulatingEvent::NeighbourInfo { .. }
-            | AccumulatingEvent::SendNeighbourInfo { .. }
-            | AccumulatingEvent::TheirKeyInfo { .. }
-            | AccumulatingEvent::TheirKnowledge { .. }
-            | AccumulatingEvent::User(_) => {
-                log_or_panic!(log::Level::Error, "unexpected event {:?}", event.payload);
-            }
-            AccumulatingEvent::Online(payload) => {
-                self.resend_bootstrap_response_join(core, &payload.p2p_node);
-            }
-        });
-    }
-
     // Finalise parsec reset and revotes for all previously unaccumulated events.
     fn finalise_parsec_reset(
         &mut self,
         core: &mut Core,
         genesis_prefix_info: GenesisPrefixInfo,
-        to_vote_again: BTreeSet<NetworkEvent>,
+        to_vote_again: Vec<NetworkEvent>,
```

Comment:
just to confirm that this changing of struct won't cause duplicate vote

Reply:
Even if we somehow passed duplicate votes to this function, they would be rejected by parsec, so there is no harm.

Reply:
ok.